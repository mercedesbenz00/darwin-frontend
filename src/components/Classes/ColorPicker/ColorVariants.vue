<template>
  <div class="color-variants">
    <div
      class="color-variants__color"
      :style="{ background: color }"
      @click="$emit('select', color)"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

export default defineComponent({
  name: 'ColorPicker',
  props: {
    color: { required: true, type: String }
  }
})
</script>

<style lang="scss" scoped>
.color-variants {
  @include col;
}

.color-variants__color {
  width: 23px;
  height: 23px;
  border-radius: $border-radius-default;
  transition: all .1s ease;
  cursor: pointer;

  &:hover {
    box-shadow: 0 0 10px 0 rgba(145,169,192,.3);
    transform: scale(1.1);
  }

  &:not(:last-child) {
    margin-bottom: 2px;
  }
}
</style>
