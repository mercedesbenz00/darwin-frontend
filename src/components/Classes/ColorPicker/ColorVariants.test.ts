import { createLocalVue, shallowMount } from '@vue/test-utils'

import ColorVariants from '@/components/Classes/ColorPicker/ColorVariants.vue'

const localVue = createLocalVue()

let propsData: { color: string }

beforeEach(() => {
  propsData = {
    color: 'rgba(255, 0, 0, 1.0)'
  }
})

it('matches snapshot', () => {
  const wrapper = shallowMount(ColorVariants, { localVue, propsData })
  expect(wrapper).toMatchSnapshot()
})

it('emit event when color is selected', () => {
  const wrapper = shallowMount(ColorVariants, { localVue, propsData })
  wrapper.findAll('.color-variants__color').at(0).trigger('click')
  expect(wrapper.emitted().select).toEqual([[propsData.color]])
})
