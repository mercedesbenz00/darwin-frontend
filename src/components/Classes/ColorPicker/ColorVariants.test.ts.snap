// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div class="color-variants">
  <div class="color-variants__color" style="background: rgb(255, 0, 0);"></div>
</div>
`;
